TUnit.Core/Executors/DedicatedThreadExecutor.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace TUnit.Core;
+
+public class DedicatedThreadExecutor : GenericAbstractExecutor
+{
+    protected sealed override async ValueTask ExecuteAsync(Func<ValueTask> action)
+    {
+        var tcs = new TaskCompletionSource<object?>();
+
+        var thread = new Thread(async void () =>
```

Comment:
I think you should stay with non-async `valueTask.AsTask.GetResult()` see https://stackoverflow.com/a/78936077/1220972

Reply:
Thanks! Good shout